ykrt/src/deopt.rs

```diff
@@ -115,6 +115,7 @@ extern "C" fn __llvm_deoptimize(
     actframes: *const c_void,
     guardid: usize,
     jitcallstack: *const c_void,
+    isswitchguard: usize,
```

Comment:
I knew you'd comment on this. This was in fact a bool at the beginning. But that made it more difficult to see if something went wrong. For example, if we read the wrong value from the argument stack, it would implicitly cast this to a bool, and the value is always `1` or `0`. Arguments are always passed as `usize` anyway, so we don't win anything by making this a `bool`. However, this being `usize` allows us to add the `debug_assert` further below, to make sure we didn't mess up reading the argument from the stack in the assembler code above.

Reply:
Then let's at least make the doc comment correct, because `true` in C does not have to evaluate to 1. In other words, we should say "if set to 0 then ... else if set to 1 then ... any other value is UB" or similar.

Reply:
Fixed in [ab1edd7](https://github.com/ykjit/yk/pull/873/commits/ab1edd7eb9ca09005b9a3ef18a9cf2ce2e24ca8b). I've also updated the type to `int` in `jitmodbuilder`.

Reply:
C's `int` and Rust's `usize` aren't the same type: you could use `intptr_t` or just `u64` (or whatever the C equivalent is).

Reply:
You're absolutely right. Fixed in [9fecddc](https://github.com/ykjit/yk/pull/873/commits/9fecddc26afe8270a05e0db29216d520d36ac6c6)